### ISSUE TITLE
Support point-in-time selection from source location

### ISSUE THREAD

**Issue by wenslayer** [cobbzilla/s3s3mirror#97]
As a system operator, I need to be able to mirror a bucket (with optional prefix) as of a recent point in time to target location (bucket + optional prefix) to satisfy an external time constraint on the restored data.

With buckets that have versioning enabled, it's possible to select files from a certain point in time. I see there is [s3-pit-restore](https://github.com/madisoft/s3-pit-restore), but it downloads files locally, whereas s3s3mirror's ability to move contents bucket to bucket is critical.

**Comment by urbien** [comment]
@wenslayer, did you find a solution? We have the same need.

**Comment by wenslayer** [comment]
@urbien, no, unfortunately not. We fell back to copying a live bucket to an interim location on a periodic basis, and then using that as our source, despite that it isn't a true point-in-time snapshot.

**Comment by cobbzilla** [comment]
@wenslayer I recently merged a change that adds support for "sync strategies".

You could define a new custom sync strategy that does exactly what you want. If you can parameterize it so that others might find it useful, it could be an interesting PR to merge into master.
